src/test/java/org/isf/opd/test/Tests.java

```diff
@@ -261,6 +334,465 @@ public void testListenerShouldUpdatePatientToMergedWhenPatientMergedEventArrive(
+				foundOpd.getNewPatient());
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(foundOpd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdListPatientId() throws Exception {
+		int code = _setupTestOpd(false);
+		Opd foundOpd = opdIoOperationRepository.findOne(code);
+		ArrayList<Opd> opds = opdBrowserManager.getOpdList(foundOpd.getPatient().getCode());
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(foundOpd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdListPatientIdZero() throws Exception {
+		Patient patient = testPatient.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+
+		Disease disease = testDisease.setup(diseaseType, false);
+		disease.setCode("angal.opd.alldisease");
+		disease.getType().setCode("angal.opd.alltype");
+
+		Opd opd = testOpd.setup(patient, disease, true);
+		GregorianCalendar now = new GregorianCalendar();
+		opd.setVisitDate(now);
+
+		patientIoOperationRepository.saveAndFlush(patient);
+		diseaseTypeIoOperationRepository.saveAndFlush(diseaseType);
+		diseaseIoOperationRepository.saveAndFlush(disease);
+		opdIoOperationRepository.saveAndFlush(opd);
+
+		ArrayList<Opd> opds = opdBrowserManager.getOpdList(0);
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(opd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdToday() throws Exception {
+		Patient patient = testPatient.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+
+		Disease disease = testDisease.setup(diseaseType, false);
+		disease.setCode("angal.opd.alldisease");
+		disease.getType().setCode("angal.opd.alltype");
+
+		Opd opd = testOpd.setup(patient, disease, true);
+		GregorianCalendar now = new GregorianCalendar();
+		opd.setVisitDate(now);
+
+		patientIoOperationRepository.saveAndFlush(patient);
+		diseaseTypeIoOperationRepository.saveAndFlush(diseaseType);
+		diseaseIoOperationRepository.saveAndFlush(disease);
+		opdIoOperationRepository.saveAndFlush(opd);
+
+		ArrayList<Opd> opds = opdBrowserManager.getOpd(false);
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(opd.getCode());
+	}
+
+	@Test
+	public void testMgrGetOpdLastWeek() throws Exception {
+		Patient patient = testPatient.setup(false);
+		DiseaseType diseaseType = testDiseaseType.setup(false);
+
+		Disease disease = testDisease.setup(diseaseType, false);
+		disease.setCode("angal.opd.alldisease");
+		disease.getType().setCode("angal.opd.alltype");
+
+		Opd opd = testOpd.setup(patient, disease, true);
+		GregorianCalendar date = new GregorianCalendar();
+		date.add(Calendar.DAY_OF_MONTH, -3);
+		opd.setVisitDate(date);
+
+		patientIoOperationRepository.saveAndFlush(patient);
+		diseaseTypeIoOperationRepository.saveAndFlush(diseaseType);
+		diseaseIoOperationRepository.saveAndFlush(disease);
+		opdIoOperationRepository.saveAndFlush(opd);
+
+		ArrayList<Opd> opds = opdBrowserManager.getOpd(true);
+		assertThat(opds.get(opds.size() - 1).getCode()).isEqualTo(opd.getCode());
+	}
```

Comment:
Also here we could had cases out of the range

Reply:
Done

Reply:
Thanks!